### PR TITLE
Missing documentation for Custom Auth

### PR DESCRIPTION
**Pull request by dplewis**
Closes: https://github.com/parse-community/Parse-SDK-JS/issues/934
Addresses: https://github.com/parse-community/Parse-SDK-JS/issues/552

Adds deprecation for _linkWith and _loginWith

I can check the links works after this PR is merged.

### CODE REVIEW

File: src/AnonymousUtils.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ const AnonymousUtils = {
    */
   logIn(options?: RequestOptions) {
     const provider = this._getAuthProvider();
-    return ParseUser._logInWith(provider.getAuthType(), provider.getAuthData(), options);
+    return ParseUser.logInWith(provider.getAuthType(), provider.getAuthData(), options);
```
Comment (acinader) [inline src/AnonymousUtils.js]: YES!! nice.

**Comment by dplewis** [comment]
@rhuanbarreto Anything you want to add?

File: src/interfaces/AuthProvider.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (acinader) [inline src/interfaces/AuthProvider.js]: why?  can you disable individual lines with the proper offending rule?  Or, can you disable the rules your concerned with here, or is there something more fundamental?  Its really good to be able to see the reason or rule in the comment....

i.e. `/* eslint-disable object-curly-spacing */`

which lets the reader know that object-curly-spacing is the rule that is an issue.

File: src/interfaces/AuthProvider.js [hunk 3]
```diff
@@ -10,28 +11,30 @@
 
 /**
  * Interface declaration for Authentication Providers
+ *
+ * @interface AuthProvider
  */
-export interface AuthProvider {
+export class AuthProvider {
```
Comment (acinader) [inline src/interfaces/AuthProvider.js]: why the change from an interface to class here, just curious.

Comment (dplewis) [reply src/interfaces/AuthProvider.js]: JSDocs didn't like it

**Comment by dplewis** [comment]
There is documentation in the guide https://docs.parseplatform.org/js/guide/#custom-authentication-module

Working code examples are in [ParseUserTest](https://github.com/parse-community/Parse-SDK-JS/blob/master/integration/test/ParseUserTest.js) and a working server setup [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/integration/server.js#L17).

**Comment by dplewis** [comment]
@acinader Perhaps we should add this to the ReadMe since this highly requested?

**Comment by rhuanbarreto** [comment]
I think that this will help more the people looking for answers and it’s enough for now. I managed to implement the Keycloak login but I’m using the REST API for now. Soon I will try to commit something showing how is it on Server side.

**Comment by dplewis** [comment]
@rhuanbarreto Having another set of eyes always helps as we tend to forget new users have to read through the documentation.

If you can try to use `linkWith` and provide feedback on if these changes are clear. We would greatly appreciate it.
